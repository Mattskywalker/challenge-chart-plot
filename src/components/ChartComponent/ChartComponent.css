.ChartComponent {

    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    height: 100%;
    background-color: #FFF;
    font-size: 1;
    
    
}

.ChartContainer {
    
    
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    
    width: 90%;
    height: 90%;
    padding: 10px;
    
    background-color: #FFF;
    
}

.Chart {
    height: 100%;
}
