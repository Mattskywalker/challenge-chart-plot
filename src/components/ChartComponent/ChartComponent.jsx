import React from 'react'
import { Chart } from "react-google-charts";

import './ChartComponent.css';

export default function ChartComponent({chartData}) {
	
    return (
        <div className='ChartComponent'>
			
			{
				chartData.rows.length > 0 &&
				<div className='ChartContainer'>
					<Chart
					className='Chart'
					chartType="Line"
					width='100%'
					height='100%'
					data={[chartData.columns, ...chartData.rows]}
					/>
				</div>
			}
			
        </div>
    )
}
