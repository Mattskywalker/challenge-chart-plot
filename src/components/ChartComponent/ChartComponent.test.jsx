import ChartComponent from ".";
import "../../setupTests";
import renderer from "react-test-renderer";

const chartData = {
    columns: ["periode", "column1", "column2"],
    rows: [
        [1, 2, 3],
        [2, 4, 2],
    ],
};
const chartDataEmpty = {
    columns: ["periode"],
    rows: [],
};

describe("ChartComponent", () => {
    it("Render chart with valid data", () => {
        const tree = renderer
            .create(<ChartComponent chartData={chartData}></ChartComponent>)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it("No Render chart with invalid or incomplete data", () => {
        const tree = renderer
            .create(
                <ChartComponent chartData={chartDataEmpty}></ChartComponent>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
