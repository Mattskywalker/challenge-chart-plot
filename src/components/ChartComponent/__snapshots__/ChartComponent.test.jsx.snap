// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartComponent No Render chart with invalid or incomplete data 1`] = `
<div
  className="ChartComponent"
/>
`;

exports[`ChartComponent Render chart with valid data 1`] = `
<div
  className="ChartComponent"
>
  <div
    className="ChartContainer"
  />
</div>
`;
